code/controllers/subsystem/SSjobs.dm

```diff
@@ -151,6 +152,12 @@ SUBSYSTEM_DEF(jobs)
 		if(player.mind && (job.title in player.mind.restricted_roles))
 			Debug("FOC incompatbile with antagonist role, Player: [player]")
 			continue
+		if(player.mind && (job.title in player.mind.single_antag_positions))
+			if(!prob(probability_of_antag_role_restriction))
+				Debug("Failed probability of getting a second antagonist position in this job, Player: [player], Job:[job.title]")
+				continue
+			else
+				probability_of_antag_role_restriction  =  probability_of_antag_role_restriction  / 10
```

Comment:
You already have `restricted_roles`  for purposes of blocking the roll of job for antag. Move this logic to `game_modes`

Reply:
I'll take a look into this tomorrow and get back to you with an update when able.

Reply:
uses .mode access now